cpp/src/gandiva/tests/projector_test.cc

```diff
@@ -1606,4 +1606,46 @@ TEST_F(TestProjector, TestCastNullableIntYearInterval) {
   EXPECT_ARROW_ARRAY_EQUALS(out_int64, outputs.at(1));
 }
 
+TEST_F(TestProjector, TestEltFunction) {
+  auto field0 = field("f0", arrow::int32());
+  auto field1 = field("f1", arrow::utf8());
+  auto field2 = field("f2", arrow::utf8());
+
+  auto schema0 = arrow::schema({field0, field1, field2});
+
+  // output fields
+  auto out_field0 = field("out_field0", arrow::utf8());
+
+  // Build expression
+  auto elt_expr0 =
+      TreeExprBuilder::MakeExpression("elt", {field0, field1, field2}, out_field0);
+
+  std::shared_ptr<Projector> projector1;
+
+  auto status = Projector::Make(schema0, {elt_expr0}, TestConfiguration(), &projector1);
+  EXPECT_TRUE(status.ok());
+
+  // Create a row-batch with some sample data
+  int num_records = 5;
+
+  auto array0 = MakeArrowArrayInt32({1, 2, 4, 0, -2}, {true, true, true, true, true});
```

Comment:
Can you add a test case for the following:
1) Return the 1 element always
2) Make the first element have size larger than the 2nd element
3) Make the first element smaller in size compared to the 2nd element

Add a test where the pos keeps changing such that:
1) elt returns the input with smallest length in the input strings
2) elt returns the input with largest length in the input strings
3) elt returns a random input

Reply:
I did not get the cases where the pos keep changing. Changing where? Do you mean a random generated pos value at execution time?

Reply:
You wrote this test - I meant changing the value of pos such that it returns the input of smallest length; input of largest length and a random input string... I believe you wrote this test.

The other test was where pos is a constant (1 for e.g.) and the first vector had the larger inputs than the 2nd vector; and another test where pos is a constant (2 for e.g.) and the 2nd vector had the smaller input

Reply:
yeah, I wrote the first test case. Now I added cases for the fixed pos where the first one grabs the pos 1 with first vector containing the bigger words, and seconde case where pos is 2 and the second vector has the smaller words. Is that what you meant?